Client/Frontend/Browser/Tab.swift

```diff
@@ -539,7 +539,30 @@ class Tab: NSObject {
 
     func setNightMode(_ enabled: Bool) {
         webView?.evaluateJavaScript("__firefox__.setNightMode(\(enabled))", completionHandler: nil)
-    }
+	}
+
+	func purgeWebView() {
+		if let webView = self.webView {
+			tabDelegate?.tab?(self, willDeleteWebView: webView)
+			webView.removeObserver(self, forKeyPath: "URL")
+			self.sessionData = self.generateSessionData()
+			self.webView = nil
+		}
+	}
+
+	func generateSessionData() -> SessionData? {
+		let currentItem: LegacyBackForwardListItem! = self.webView?.backForwardList.currentItem
+		
+		if currentItem != nil {
```

Comment:
why not to use `if let currentItem = self.webView?.backForwardList.currentItemt`